x-pack/apm-server/sampling/pubsub/pubsub.go

```diff
@@ -141,9 +162,11 @@ func (p *Pubsub) searchTraceIDs(ctx context.Context, out chan<- string, observed
 		indices = append(indices, index)
 	}
 	if err := p.refreshIndices(ctx, indices); err != nil {
-		return err
+		return false, err
 	}
 
+	var changed bool
+	var observedSeqnosMu sync.Mutex
 	g, ctx := errgroup.WithContext(ctx)
 	for _, index := range indices {
```

Comment:
Unrelated to this PR, but while I'm looking: do you have an estimate how many indices will typically be queried? Wondering about how many requests this could trigger `apm-server` to send out at once, if that's a concern.

Reply:
_Probably_ only ever one or two. We haven't set up ILM for this yet, but the idea is to have a fairly aggressive rollover + deletion policy: rollover the index after (say) an hour, and then delete after another hour. I say probably because we could also have size based rollover, i.e. rollover when the index grows to 30GB. I don't think we'll need that, but we'll see.

Whatever rollover/delete time duration we choose defines the maximum duration for a trace. If an event for a trace comes in a day later, then the sampled trace ID will have been long forgotten, so the event will end up getting dropped.